R CODE FINAL Alexander Shemetev

```diff
@@ -0,0 +1,791 @@
+#    of the packages mentioned in the course)
+matrix_raw <- as.matrix(raw_data)
+heatmap(matrix_raw)
+# Heatmap successfully created
+
+# 9) Create a 2x4 matrix of boxplots where x = default, y is all other columns (all 8 plots should be on one picture)
+# I did not quite clear understand the task. 
+# Here is simple boxplot for columns: 
+boxplot(raw_data, main = "boxplot.matrix(...., main = raw_data$default)",
+        notch = TRUE, col = 2:4)
+# Here is ggplot boxplot: 
+require(reshape2)
+library(ggplot2)
+library(gridExtra)
+library(MASS)
+# Compiled from official ggplot-2 documentation
+df <- subset(raw_data, select=c(default, age, ed, employ, address, income, debtinc, creddebt, othdebt))  
+p <- list(df)
+for (j in colnames(df)) {
+  p[[j]] <- ggplot(data=df, aes_string(x=df$default, y=j)) + # Specify dataset, input or grouping col name and Y
+    geom_boxplot(aes(fill=factor(type))) + guides(fill=FALSE) + # Boxplot by which factor + color guide
+    theme(axis.title.y = element_text(face="bold", size=14))  # Make the Y-axis labels bigger/bolder
+  
+}
+
+do.call(grid.arrange, c(p, ncol=2))
+
+
+# Compiled myself for resolving this task (METHOD 2)
+bp1 <- ggplot(raw_data, aes(x=default, y=age, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp1
+bp2 <- ggplot(raw_data, aes(x=default, y=ed, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp2 
+bp3 <- ggplot(raw_data, aes(x=default, y=employ, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp3 
+bp4 <- ggplot(raw_data, aes(x=default, y=address, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp4 
+bp5 <- ggplot(raw_data, aes(x=default, y=income, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp5 
+bp6 <- ggplot(raw_data, aes(x=default, y=debtinc, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp6 
+bp7 <- ggplot(raw_data, aes(x=default, y=creddebt, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp7 
+bp8 <- ggplot(raw_data, aes(x=default, y=othdebt, group=default)) + 
+  geom_boxplot(aes(fill=default))
+bp8 
+plist <- list(bp1, 	bp2, 	bp3, 	bp4, 	bp5, 	bp6, 	bp7, 	bp8)
+do.call("grid.arrange", c(plist))
+do.call(grid.arrange, c(plist, ncol=2))
+
+# SUCCESS - 2X4 matrix of the BOXPLOTS CREATED on 1 picture 8 charts. 
+# If there is a long time expectation - please, re-run the last line again: 
+#do.call(grid.arrange, c(plist, ncol=2))
+# It creates 2X4 boxplot matrix
+
+
+
+
+
+# 10) Create histogram of employ for those who defaulted (default = 1) and not (default  = 0) on the same graph (use color to distinguish 
+#     defaulted and nondefaulted). Make histogram a bit transparent by setting alpha to 0.5 (hope you remember how to do it)
+
+# version 3: data divided by 2 columns: 1 and 0 default; alpha 0.5; coloring by the number of employed (decoration)
+bpe3 = ggplot(raw_data,aes(x=raw_data$default,group=employ,fill=employ))+
+  geom_histogram(position="identity",alpha=0.5,binwidth=0.25)+theme_bw()
+bpe3 
+
```

Comment:
Create histogram of employ for those who defaulted (default = 1) and not (default  = 0) implies employ has to be on x-axis